corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -0,0 +1,24 @@
+var COMMCAREHQ_MODULES = {};
+
+function hqDefine(path, moduleObject) {
+    var parts = ['COMMCAREHQ_MODULES'].concat(path.split('/'));
+    var i;
+    var object = window;
+    for (i = 0; i < parts.length - 1; i += 1) {
+        if (typeof object[parts[i]] === 'undefined') {
+            object[parts[i]] = {};
+        }
+        object = object[parts[i]];
+    }
+    object[parts[i]] = moduleObject;
```

Comment:
Am I understanding correctly that this will overwrite an already-defined module if two modules are defined with the same path? Seems like that could happen pretty easily if someone copy/pasted a module from one file to another and then forgot to update the path. Should we check for that and throw an error if it happens?